test/parallel_api/iterator/tuple_unit.pass.cpp

```diff
@@ -0,0 +1,191 @@
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include _PSTL_TEST_HEADER(tuple)
+
+#include "support/utils.h"
+
+template <typename... _T>
+static oneapi::dpl::__internal::tuple<_T...>
+to_onedpl_tuple(const std::tuple<_T...>& __t)
+{
+    return oneapi::dpl::__internal::tuple<_T...>(__t);
+}
+
+template <typename Tuple1, typename Tuple2>
+void
+test_tuple(Tuple1 t1, Tuple2 t2)
+{
+
+    auto onedpl_t1 = to_onedpl_tuple(t1);
+    auto onedpl_t2 = to_onedpl_tuple(t2);
+
+    static_assert(std::is_trivially_copyable_v<decltype(onedpl_t1)>, "oneDPL tuple is not trivially copyable");
+
+    // Test binary comparison operators for std::tuple and oneAPI::dpl::__internal::tuple
+    EXPECT_EQ((t1 == t2), (onedpl_t1 == onedpl_t2), "equality comparison does not match std::tuple");
+    EXPECT_EQ((t1 != t2), (onedpl_t1 != onedpl_t2), "inquality comparison does not match std::tuple");
+    EXPECT_EQ((t1 < t2), (onedpl_t1 < onedpl_t2), "less than comparison does not match std::tuple");
+    EXPECT_EQ((t1 <= t2), (onedpl_t1 <= onedpl_t2), "less than or equal to comparison does not match std::tuple");
+    EXPECT_EQ((t1 > t2), (onedpl_t1 > onedpl_t2), "greater than comparison does not match std::tuple");
+    EXPECT_EQ((t1 >= t2), (onedpl_t1 >= onedpl_t2), "greater than or equal to comparison does not match std::tuple");
+
+    EXPECT_EQ((t1 == t2), (t1 == onedpl_t2), "equality comparison does not match std::tuple");
+    EXPECT_EQ((t1 != t2), (t1 != onedpl_t2), "inquality comparison does not match std::tuple");
+    EXPECT_EQ((t1 < t2), (t1 < onedpl_t2), "less than comparison does not match std::tuple");
+    EXPECT_EQ((t1 <= t2), (t1 <= onedpl_t2), "less than or equal to comparison does not match std::tuple");
+    EXPECT_EQ((t1 > t2), (t1 > onedpl_t2), "greater than comparison does not match std::tuple");
+    EXPECT_EQ((t1 >= t2), (t1 >= onedpl_t2), "greater than or equal to comparison does not match std::tuple");
+
+    EXPECT_EQ((t1 == t2), (onedpl_t1 == t2), "equality comparison does not match std::tuple");
+    EXPECT_EQ((t1 != t2), (onedpl_t1 != t2), "inquality comparison does not match std::tuple");
+    EXPECT_EQ((t1 < t2), (onedpl_t1 < t2), "less than comparison does not match std::tuple");
+    EXPECT_EQ((t1 <= t2), (onedpl_t1 <= t2), "less than or equal to comparison does not match std::tuple");
+    EXPECT_EQ((t1 > t2), (onedpl_t1 > t2), "greater than comparison does not match std::tuple");
+    EXPECT_EQ((t1 >= t2), (onedpl_t1 >= t2), "greater than or equal to comparison does not match std::tuple");
+
+    auto onedpl_t3 = to_onedpl_tuple(t1);
+    auto onedpl_t4 = to_onedpl_tuple(t2);
```

Comment:
Maybe `decltype(onedpl_t1) onedpl_t3{};` etc. - to ensure that the assignment really does the work, and not just "duplicates" construction?

Reply:
Maybe I'm missing the point of your comment, but the highlighted lines are not the assignment being tested below.  These lines construct two onedpl internal tuples, and the subsequent two lines construct two std::tuples.  Then the later lines immediately preceding the `EXPECT_TRUE` calls are the assignment.

Reply:
Currently, this is what happens with `onedpl_t3`:

```
    auto onedpl_t3 = to_onedpl_tuple(t1);
...
    onedpl_t3 = t1;
```
So the assignment sets essentially the same value as the constructor. That seems redundant at the very least. Since you intend to test the assignment, the value at construction should not matter, so for me it would be best to set no specific value. That would additionally test that the internal tuple is default-constructible/

Reply:
Sure, that makes sense.  
The intention here was to do 
    auto onedpl_t3 = to_onedpl_tuple(t1);
...
    onedpl_t3 = t1;

Reply:
done.